lib: respect terminal capabilities on styleText (#54389)
nodejs/node: doc/api/util.md

// Prints "value"
```

## `util.styleText(format, text[, options])`

> Stability: 1.1 - Active development

<!-- YAML
added:
  - v21.7.0
  - v20.12.0
changes:
  - version: REPLACEME
    pr-url: https://github.com/nodejs/node/pull/54389
    description: Respect isTTY and environment variables
      such as NO_COLORS, NODE_DISABLE_COLORS, and FORCE_COLOR.
-->

* `format` {string | Array} A text format or an Array
  of text formats defined in `util.inspect.colors`.
* `text` {string} The text to to be formatted.
* `options` {Object}
  * `validateStream` {boolean} When true, `stream` is checked to see if it can handle colors. **Default:** `true`.
  * `stream` {Stream} A stream that will be validated if it can be colored. **Default:** `process.stdout`.

This function returns a formatted text considering the `format` passed

> for printing in a terminal, it is aware of the terminal's capabilities

Reviewer: I would totally understand not wanting to run the CI again just because of this change, so totally fine with me if this lands without this suggestion. If someone wants an easy doc fix for a subsequent PR, though, here it is.

```suggestion
for printing in a terminal. It is aware of the terminal's capabilities
```

added:
  - v21.7.0
  - v20.12.0
changes:
  - version: REPLACEME
    pr-url: https://github.com/nodejs/node/pull/54389
    description: Respect isTTY and colors environment variables
      such as NO_COLORS, NODE_DISABLE_COLORS, and FORCE_COLOR.
-->

* `format` {string | Array} A text format or an Array
  of text formats defined in `util.inspect.colors`.
* `text` {string} The text to to be formatted.

> * `options` {Object}

Reviewer 2: You need to add this to the function's signature above.

Author: Done in [cc08b60](https://github.com/nodejs/node/pull/54389/commits/cc08b60053766f5224d2aa95fdc27135e1933412)

Reviewer 2: It's not what I meant. You have to update https://github.com/nodejs/node/blob/22daeba24d0dacc2d9377d9c46c562da27260f9e/doc/api/util.md#L1805

Reviewer 3: @RafaelGSS I've updated the doc for ya :)

added:
  - v21.7.0
  - v20.12.0
changes:
  - version: REPLACEME
    pr-url: https://github.com/nodejs/node/pull/54389

>     description: Respect isTTY and colors environment variables

Reviewer 4: ```suggestion
    description: Respect isTTY and environment variables
```

      such as NO_COLORS, NODE_DISABLE_COLORS, and FORCE_COLOR.
-->

* `format` {string | Array} A text format or an Array
  of text formats defined in `util.inspect.colors`.
* `text` {string} The text to to be formatted.
* `options` {Object}
  * `validateStream` {boolean} When true, `stream` is checked to see if it can handle colors. **Default:** `true`.
  * `stream` {Stream} A stream that will be validated if it can be colored. **Default:** `process.stdout`.

> This function returns a formatted text considering the `format` passed
> for printing in a terminal, it is aware of the terminal's capabilities
> and act according to the configuration set via `NO_COLORS`,
> `NODE_DISABLE_COLORS` and `FORCE_COLOR` environment variables.

Reviewer 4: ```suggestion
This function returns a formatted string based on the `format` passed.
When `options.validateStream` is `true`, the `NO_COLORS`, `NODE_DISABLE_COLORS`,
and `FORCE_COLOR` environment variables, along with the input `stream`,
are used to determine whether colors are supported. If they are not, the string
is returned unchanged.
```
